content/for-developers/tutorials/inner-loop/05-expose-app/expose-app.md

```diff
@@ -1,99 +1,76 @@
+- Create route
 
-b. Apply the configuration to create the route:
+- Create service based on the application requirements
 
-`oc apply -f route.yaml`
-c. Once the route is created, it will assign a hostname (`nordmart.example.com`) that you can use to access your application externally.
+## Tutorial
 
-Note: In order to use a hostname, you must have a DNS record pointing to the OpenShift cluster's external IP or load balancer.
+Assuming that you already have deployed the application using the previous tutorial.
 
-By following the steps outlined above, you can successfully expose your application deployed on Red Hat OpenShift, using the example application "`nordmart`". Whether using the NodePort, LoadBalancer, or Route approach, you can make your application accessible from outside the OpenShift cluster, allowing users to access and interact with your application over the network.
+OpenShift provides a routing mechanism called "routes" that allows you to expose applications using hostnames and paths. Routes are created using the Route resource and can provide additional features such as SSL termination and path-based routing. To create a route:
 
-## Service Type
+1. In your `tilt/values-local` file, define a route:
 
-In OpenShift, services are used to expose applications internally within the cluster. To expose your application externally, you need to specify the appropriate service type:
+    ```yaml
+    ## Route
+    route:
+      enabled: true
+      port:
+        targetPort: http
+    ```
 
-a. In your deployment configuration YAML file, define a service with the type NodePort or LoadBalancer:
+    a) Route is `enabled: true`, which means the route resource will be created when you tilt up.
 
-```yaml
-apiVersion: v1
-kind: Service
-metadata:
-  name: nordmart-service
-spec:
-  type: NodePort
-  ports:
-    - port: 80
-      targetPort: 8080
-  selector:
-    app: nordmart
-```
+    b) `port:` This section defines the port configuration for the Route resource.
 
-In the above example, a service named `nordmart-service` is defined with the NodePort type, exposing port 80 and forwarding traffic to port 8080 of the application.
+    c) `targetPort: http`: This specifies the target port for the Route. In this case, the value is set to `http`, which is a named port defined in the associated Service configuration. It represents the port on which the backend service is listening to handle incoming traffic.
 
-### NodePort
+    Route is responsible for routing the external traffic based on rules while a Service is responsible for load balancing and forwarding the traffic to the appropriate pods. To create a service:
 
-When using the NodePort service type, OpenShift assigns a random port within a predefined range on each node of the cluster. To access your application externally, you need to determine the assigned port and the IP address of any of the cluster's nodes:
+1. In your `tilt/values-local` file, define a service:
```

Comment:
Expose Application Internally 
- define a service first 
- visit ui to see the service 


Expose Application Externally via Route
- define a route in values
- visit the ui 
- visit the route

Expose Application Externally via Ingress
- define a ingress in values
- visit the ui ingress
- see the route generated by ingress
- visit the route

Reply:
we can proceed through the tutorial using this flow

Reply:
Done